### PR TITLE
Batch fetching storage values again

### LINKED ISSUE #1195: Regression in speed of fetching child storage values in 0.32
As I understand, when iterating over a map in the storage, this constant
`const STORAGE_FETCH_PAGE_SIZE: u32 = 32;`
is used to determine how many keys should be fetched at once. 

This being a constant is quite limiting -- in fact for maps of size a few thousands, this becomes slow because of this constant being too small. Ideally it should be possible to pass this into `iter()` or somewhere, so that one can configure the page size depending on factors like size of the key and value, instead of using a constant.

### PR DESCRIPTION
**Pull request by jsdw**
When the Backend trait was added, I moved to fetching storage values one by one because the batch function we used was a little confusing and it seemed simpler at the time.

This PR adds back the fetching of storage values in batches, and improves the docs on the relevant RPC methods to clarify their behaviour.

Closes #1195

### CODE REVIEW

**Review by lexnv — APPROVED** [review]

**Review by tadeohepperle — APPROVED** [review]

**Review by niklasad1 — APPROVED** [review]
LGTM

Makes sense to me that you removed it in the first place because the code is quite complicated

**Comment by jsdw** [comment]
Before I merge this I'll just see if its worth adding a test or something to make sure that the storage values returned are as expected.

**Pull request merged**
